client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/utils/BsonHelper.java

```diff
@@ -0,0 +1,136 @@
+            Method toBsonDocument = query.getClass().getMethod("toBsonDocument");
+            Object bsonDocument = toBsonDocument.invoke(query);
+
+            Object documentCodec = documentCodecClass().getDeclaredConstructor().newInstance();
+            Method asBsonReader = bsonDocumentClass().getMethod("asBsonReader");
+            Method builder = decoderContextClass().getMethod("builder");
+            Object builderInstance = builder.invoke(null);
+            Method build = builderInstance.getClass().getMethod("build");
+
+            Method decode = documentCodecClass().getMethod("decode", bsonReaderClass(), decoderContextClass());
+            return decode.invoke(documentCodec, asBsonReader.invoke(bsonDocument), build.invoke(builderInstance));
+        } catch (NoSuchMethodException | InstantiationException | IllegalAccessException |
+                 InvocationTargetException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static String getType(Object bsonType) {
+        try {
+            Class<?> bsonTypeClassMapClass = Class.forName("org.bson.codecs.BsonTypeClassMap");
+            Class<?> bsonTypeClass = Class.forName("org.bson.BsonType");
+            Object bsonTypeClassMap = bsonTypeClassMapClass.getDeclaredConstructor().newInstance();
+            Method get = bsonTypeClassMapClass.getMethod("get", bsonTypeClass);
+            Object type = get.invoke(bsonTypeClassMap, bsonType);
+            return (String) type.getClass().getMethod("getTypeName").invoke(type, null);
+        } catch (ClassNotFoundException | NoSuchMethodException | InstantiationException | IllegalAccessException |
+                 InvocationTargetException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Object getTypeFromNumber(Integer number) {
+        Class<?> bsonTypeClass;
+        try {
+            bsonTypeClass = Class.forName("org.bson.BsonType");
+            Method findByValue = bsonTypeClass.getMethod("findByValue", int.class);
+            return findByValue.invoke(null, number);
+        } catch (ClassNotFoundException | IllegalAccessException | InvocationTargetException |
+                 NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Object getTypeFromAlias(String alias) {
+        Class<?> bsonTypeClass;
+        try {
+            bsonTypeClass = Class.forName("org.bson.BsonType");
+            Method valueOf = bsonTypeClass.getMethod("valueOf", String.class);
+            return valueOf.invoke(null, alias);
+        } catch (ClassNotFoundException | IllegalAccessException | InvocationTargetException |
+                 NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Class<?> documentClass() {return getClass("org.bson.Document");}
+
+    private static Class<?> bsonDocumentClass() {return getClass("org.bson.BsonDocument");}
```

Comment:
for performance reasons, these could all be stored directly in static variables, instead of being searched at each use. however, would need to make sure to do not crash the controller for SUTs that do not use MongoDB

Reply:
Done